developers/mlt++.md

```diff
@@ -0,0 +1,566 @@
+- mlt_factory_link        ==> Mlt::Factory::link
+- mlt_factory_transition  ==> Mlt::Factory::transition
+- mlt_factory_consumer    ==> Mlt::Factory::consumer
+- mlt_factory_close       ==> Mlt::Factory::close
+
+NB: Factory usage for service construction is optional.
+
+Types:
+- mlt_chain               ==> Mlt::Chain
+- mlt_properties          ==> Mlt::Properties
+- mlt_frame               ==> Mlt::Frame
+- mlt_link                ==> Mlt::Link
+- mlt_service             ==> Mlt::Service
+- mlt_producer            ==> Mlt::Producer
+- mlt_filter              ==> Mlt::Filter
+- mlt_transition          ==> Mlt::Transition
+- mlt_consumer            ==> Mlt::Consumer
+
+Methods:
+
+- mlt_type_method         ==> Mlt::Type.method
+- ie: mlt_playlist_append     ==> Mlt::Playlist.append
+
+Parent methods are available directly on children.
+
+Additionally, you can specify:
+
+~~~
+using namespace Mlt;
+~~~
+
+To avoid the enforced use of the Mlt:: prefix.
+
+Enumerators and macros are reused directly from the C library.
+
+#### Class Hierarchy
+
+The currently mapped objects are shown in the following hierarchy:
+
+<pre>
+Factory
+  Properties
+    Frame
+    Service
+    Consumer
+    Field
+    Filter
+    Multitrack
+    Producer
+      Chain
+      Link
+      Playlist
+      Tractor
+      Transition
+</pre>
+
+#### Special Cases
+
+Care should be taken with wrapper objects. 
+
+Taking, as an example, the C function that returns the immediate consumer of
+a service:
+
+~~~
+mlt_service mlt_service_consumer( mlt_service );
+~~~
+
+This maps to:
+
+~~~
+Mlt::Service *Mlt::Service.consumer( );
+~~~
+
+Note that you get an object back - it is never the original c++ object, but 
+a wrapping object. This is done to keep consistency with the C api which may 
+instantiate C instances - therefore it cannot be assumed that a C++ object 
+exists for all mlt service instances.
+
+As such, it is mandatory that you delete these objects. The original will 
+not be affected. However, all other modifications (to properties or its
+state of connection) will be reflected in the original object.
+
+This approach excludes the use of RTTI to determine the real type of the 
+object - this can only be done by parsing the objects properties.
+
+Objects may be invalid - always use the is_valid method to check validity 
+before use.
+
+#### Limitations
+
+The mechanisms for the definition of new services are deliberately 
+excluded from the C++ wrappings - this is done to ensure that service 
+networks constructed can be serialized and used by existing applications
+which are based on the C API (such as melted).
+
+#### Hello World
+
+The mlt++ wrapper is a c++ wrapper for the mlt C library. As such, it 
+provides clean C++ access to the underlying library.
+
+An example of use is as follows:
+
+~~~
+#include <mlt++/Mlt.h>
+using namespace Mlt;
+
+int main( void )
+{
+    Factory::init( );
+    Producer p( "pango:", "Hello World" );
+    Consumer c( "sdl" );
```

Comment:
Where did this document come from? This example is no longer correct.

Reply:
I migrated this file - which I think we could remove now:
https://github.com/mltframework/mlt/blob/master/docs/mlt%2B%2B.txt

Fixed